src/common/hooks/useRestoreToken.tsx

```diff
@@ -44,34 +44,66 @@ export const useRestoreToken = (
   const [errorMessage, setErrorMessage] = useState<string>();
   const [state, setState] = useState<ContractFunctionState>("UNINITIALIZED");
 
+  const sendToTitleEscrow = async (previousBeneficiary: string, previousHolder: string): Promise<void> => {
+    if (!tokenId) throw new Error("Ownership data is not provided");
+    if (!contractInstance?.address) throw new Error("Token Registry Instance should have address");
+    setState("PENDING_CONFIRMATION");
+
+    const sendToNewEscrowReceipt = await contractInstance?.sendToNewTitleEscrow(
+      previousBeneficiary,
+      previousHolder,
+      tokenId
+    );
+
+    const sendToNewEscrowTx = await sendToNewEscrowReceipt.wait();
+
+    const deployedTitleEscrowArgs = sendToNewEscrowTx.events?.find((event) => event.event === "TitleEscrowDeployed")
+      ?.args;
+
+    if (!deployedTitleEscrowArgs || !deployedTitleEscrowArgs[0])
+      throw new Error(`Address for deployed title escrow cannot be found. Tx: ${JSON.stringify(sendToNewEscrowTx)}`);
+
+    const sendTokenArgs = sendToNewEscrowTx.events?.find((event) => event.event === "Transfer")?.args;
+    if (!sendTokenArgs)
+      throw new Error(`Token was not restored to owner and beneficiary. Tx: ${JSON.stringify(sendToNewEscrowTx)}`);
```

Comment:
Does this really mean that the token was not restored to owner and bene? It suggest that there is no transfer event (from the erc721 contract). It's existence will not say that its restored to the owner/bene. 

Can either:
1. Trust the contract to work correctly and skip these checks (hidden coupling)
2. Validate the title escrow deployed has the right bene & holder

Reply:
ok have added the validation to check if the new title escrow being restored to consists of the previousBeneficiary and previousHolder. this validation is being used in both sendToTitleEscrow and deployAndSendToTitleEscrow.